lib/SDIR/Translate/translateToSDFG.cpp

```diff
@@ -0,0 +1,1597 @@
+      map.setEntryID(nodeID++);
+      map.setExitID(nodeID++);
+      if (translateMapToSDFG(map, jemit).failed())
+        return failure();
+    }
+
+    if (ConsumeNode consume = dyn_cast<ConsumeNode>(oper)) {
+      consume.setEntryID(nodeID++);
+      consume.setExitID(nodeID++);
+      if (translateConsumeToSDFG(consume, jemit).failed())
+        return failure();
+    }
+  }
+
+  jemit.endList(); // nodes
+
+  jemit.startNamedList("edges");
+
+  for (Operation &oper : op.body().getOps()) {
+    if (CopyOp edge = dyn_cast<CopyOp>(oper))
+      if (translateToSDFG(*edge, jemit).failed())
+        return failure();
+
+    if (StoreOp edge = dyn_cast<StoreOp>(oper))
+      if (translateToSDFG(*edge, jemit).failed())
+        return failure();
+
+    if (sdir::CallOp edge = dyn_cast<sdir::CallOp>(oper))
+      if (translateToSDFG(*edge, jemit).failed())
+        return failure();
+  }
+
+  jemit.endList(); // edges
+
+  jemit.endObject();
+  return success();
+}
+
+//===----------------------------------------------------------------------===//
+// TaskletNode
+//===----------------------------------------------------------------------===//
+
+// Temporary auto-lifting. Will be included into DaCe
+LogicalResult liftToPython(TaskletNode &op, JsonEmitter &jemit) {
+  int numOps = 0;
+  Operation *firstOp = nullptr;
+
+  for (Operation &oper : op.body().getOps()) {
+    if (numOps >= 2)
+      return failure();
+    if (numOps == 0)
+      firstOp = &oper;
+    ++numOps;
+  }
+
+  AsmState state(op);
+
+  if (dyn_cast<arith::AddFOp>(firstOp) || dyn_cast<arith::AddIOp>(firstOp)) {
+    std::string nameArg0;
+    llvm::raw_string_ostream nameArg0Stream(nameArg0);
+    op.getArgument(0).printAsOperand(nameArg0Stream, state);
+    nameArg0.erase(0, 1); // Remove %-sign
+
+    std::string nameArg1;
+    llvm::raw_string_ostream nameArg1Stream(nameArg1);
+    op.getArgument(1).printAsOperand(nameArg1Stream, state);
+    nameArg1.erase(0, 1); // Remove %-sign
+
+    jemit.printKVPair("string_data", "__out = " + nameArg0 + " + " + nameArg1);
+    jemit.printKVPair("language", "Python");
+    return success();
+  }
+
+  if (dyn_cast<arith::MulFOp>(firstOp) || dyn_cast<arith::MulIOp>(firstOp)) {
+    std::string nameArg0;
+    llvm::raw_string_ostream nameArg0Stream(nameArg0);
+    op.getArgument(0).printAsOperand(nameArg0Stream, state);
+    nameArg0.erase(0, 1); // Remove %-sign
+
+    std::string nameArg1;
+    llvm::raw_string_ostream nameArg1Stream(nameArg1);
+    op.getArgument(1).printAsOperand(nameArg1Stream, state);
+    nameArg1.erase(0, 1); // Remove %-sign
+
+    jemit.printKVPair("string_data", "__out = " + nameArg0 + " * " + nameArg1);
+    jemit.printKVPair("language", "Python");
+    return success();
+  }
+
+  if (arith::ConstantFloatOp oper = dyn_cast<arith::ConstantFloatOp>(firstOp)) {
+    std::string val = std::to_string(oper.value().convertToDouble());
+    jemit.printKVPair("string_data", "__out = dace.float64(" + val + ")");
```

Comment:
assuming the type here is wrong, you can use the return type. (what if I defined a float32 constant?)